_posts/2016-07-03-handling-big-projects.markdown

```diff
@@ -0,0 +1,123 @@
+
+To support dynamic content on the website, I have a rake command to use [a decade old][yard] documentation formatter to pull content from a [13 year old][rubygems] dependency manager - that lets others describer their project's. In order to let them keep it up to date, I have a tiny 35 line web-server that allows specific projects to trigger a new CI build.
+
+### Plugins
+
+<blockquote class="twitter-tweet" data-lang="en-gb"><p lang="en" dir="ltr">Summary of every big OSS project. Monolith -&gt; Plugin support.<a href="https://t.co/7x4vuW4bRd">https://t.co/7x4vuW4bRd</a></p>&mdash; Orta Therox (@orta) <a href="https://twitter.com/orta/status/748561323164864512">30 June 2016</a></blockquote> <script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+It's almost inevitable that once a project becomes big, maintainers have to become a lot more conservative about how they introduce new code. You become good at saying 'no', but that a lot of people have legitimate needs. So, instead end up building your tool into a platform.
+
+Some of the most interesting ideas in the CocoaPods ecosystem comes from plugins.
+
+I wanted plugins to be a first class citizen within Danger from day one. It's in the [VISION][vision] file, and it's applied into how I've designed a lot of the user-facing site. I was [torn][plugins_1] after a few months of development ][where things][plugins_2] should go. Now the core of Danger is [built as plugins][plugins_3].
+
+### Documentation
+
+My second big project on CocoaPods was collating documentation and scoping different types of documentation. In CocoaPods I ended with:
+
+* **Highlight pages** (intro pages, app pages, team pages)
+* **Guides** (tutorials, overviews, FAQs)
+* **Reference** (Command-line interface, APIs for developers)
+
+These 3 buckets for documentation makes it pretty easy to separate where people should look depending on what they're looking for. This pattern I'm stealing outright for Danger. Just not quite there yet, it's a blocker on 1.0 though.
+
+## User Data
+
+Not storing any, phew! Though if [Danger as a Service][daas] happens, then it will.
+
+## People
+
+People are hard, Ash said in [Building Popular Projects][building_pop_projects]:
+
+> The biggest existential threat to your library is this: you get burned out and stop working on it – and no one else contributes to it –
+
+Understanding motivations, encouraging ownership and accommodating multiple viewpoints are vital parts of anyone who wants to make a project bigger than themselves. There [are lots of times][danger_contributions] when I'm not the lead developer on Danger.
+
+I owe a lot of this to the policy Ash and I created with Moya, the wordy "[Moya Community Continuity Guidelines][moya_guidelines]" which defines the expectations for the maintainers of a project towards contributors.
```

Comment:
"which define**s** the" -> "which define the", unless this is one of those cases where English English and North American English differ on pluralization.